common/src/chain/block/block_index.rs

```diff
@@ -0,0 +1,75 @@
+use crate::chain::block::Block;
+use crate::chain::ChainConfig;
+use crate::primitives::{BlockHeight, Id, Idable};
+// use crate::Uint256;
+use parity_scale_codec::{Decode, Encode};
+
+#[derive(Debug, Clone, PartialEq, Eq, Encode, Decode)]
+#[allow(dead_code, unused_variables)]
+pub struct BlockIndex {
+    block_id: Id<Block>,
+    prev_block_id: Option<Id<Block>>,
+    next_block_id: Option<Id<Block>>,
+    // TODO: When Carla finish her code, we should use Uint256 at the moment it's unable to store to DB
+    //  pub chain_trust: Uint256,
+    chain_trust: u128,
+    height: BlockHeight,
+    time: u32,
```

Comment:
As I noted elsewhere, it would be nice to have a separate type for time rather than using `u32`

Reply:
This is actually a very confusing discussion that we can have because pleasing everyone is gonna be hard. On one hand, we still have like 80 years before this overflows; this is a point @altonen brought up that u32 may not be enough. On the other hand, handling time is one of the nastiest things we can ever deal with, which is why programming languages have specific types for time that are abstract and can be converted to any integer type with any precision and unit, such as chrono in C++ and Rust. I'm heavily concerned that this can very easily blow up to a confusing type whose serialization will become a nightmare in different contexts. So, I suggest to open a github issue about this and study it later instead of going on a tangent right now about it.

Anton or Lukas. Once you open an issue, please resolve this discussion.

Reply:
Done! https://github.com/mintlayer/mintlayer-core/issues/127#issue-1187630040